index.js

```diff
@@ -258,106 +361,212 @@ export const PullRequestCollection = {
 
   if (index > 0) {
     // Process all new events in oldest-to-newest order
-    const newEvents = data.slice(0, index).reverse();  
+    const newEvents = data.slice(0, index).reverse();
     for (let event of newEvents) {
       const { type, payload } = event;
-      for (let event of state.repos[key].events) {
+      for (let eventData of state.repos[key].events) {
+        const [event, number] = split(eventData, '/');
         switch (event) {
-          case "issueOpened": {
-            if (type === "IssuesEvent" && payload.action === "opened") {
+          case 'issueOpened': {
+            if (type === 'IssuesEvent' && payload.action === 'opened') {
               const repoRef = root.users.one({ name: owner }).repos.one({ name: repo });
               const number = payload.issue.number;
               await repoRef.issueOpened.dispatch({
-                issue: repoRef.issues.one({ number })
+                issue: repoRef.issues.one({ number }),
+              });
+            }
+            break;
+          }
+          case 'issueClosed': {
+            if (type === 'IssuesEvent' && payload.action === 'closed') {
+              const issueRef = root.users
+                .one({ name: owner })
+                .repos.one({ name: repo })
+                .issues.one({ number: number });
+              await issueRef.closed.dispatch();
+            }
+            break;
+          }
+          case 'releasePublished': {
+            if (type === 'ReleaseEvent' && payload.action === 'published') {
+              const repoRef = root.users.one({ name: owner }).repos.one({ name: repo });
+              const id = payload.release.id;
+              await repoRef.releasePublished.dispatch({
+                release: repoRef.releases.one({ id }),
               });
             }
             break;
           }
-          case "pullRequestOpened": {
-            if (type === "PullRequestEvent" && payload.action === "opened") {
+          case 'pullRequestOpened': {
+            if (type === 'PullRequestEvent' && payload.action === 'opened') {
               const repoRef = root.users.one({ name: owner }).repos.one({ name: repo });
               const number = payload.pull_request.number;
               await repoRef.pullRequestOpened.dispatch({
                 issue: repoRef.issues.one({ number }),
-                pullRequest: repoRef.pullRequests.one({ number })
+                pullRequest: repoRef.pullRequests.one({ number }),
               });
             }
             break;
           }
+          case 'pullRequestClosed': {
+            if (type === 'PullRequestEvent' && payload.action === 'closed') {
+              const pullRef = root.users
+                .one({ name: owner })
+                .repos.one({ name: repo })
+                .pullRequests.one({ number: number });
+              await pullRef.closed.dispatch();
+            }
+            break;
+          }
+          case 'pullRequestMerged': {
```

Comment:
Will this case ever be run? Seems like the closed case above will always take precedence. Let's remove the `merged` event, it's better to check in the handler the status of the sender